tests/unit_tests/test_matchmaker_algorithm_team_matchmaker.py

```diff
@@ -0,0 +1,248 @@
+import logging
+
+import pytest
+
+from server import config
+from server.matchmaker import CombinedSearch, Search
+from server.matchmaker.algorithm.team_matchmaker import (
+    TeamMatchMaker,
+    UnevenTeamsException
+)
+
+
+@pytest.fixture(scope="module")
+def player_factory(player_factory):
+    player_id_counter = 0
+
+    def make(
+        mean: int = 1500,
+        deviation: int = 500,
+        ladder_games: int = config.NEWBIE_MIN_GAMES+1,
+        name=None
+    ):
+        nonlocal player_id_counter
+        """Make a player with the given ratings"""
+        player = player_factory(
+            ladder_rating=(mean, deviation),
+            ladder_games=ladder_games,
+            login=name,
+            lobby_connection_spec=None,
+            player_id=player_id_counter
+        )
+        player_id_counter += 1
+        return player
+    return make
+
+
+def make_searches(ratings, player_factory):
+    searches = []
+    for i, r in enumerate(ratings):
+        searches.append(Search([player_factory(r + 300, 100, name=f"p{i}")]))
+    return searches
+
+
+def test_team_matchmaker_performance(player_factory, bench, caplog):
```

Comment:
Not sure how good these performance tests are. Maybe we should put them behind a marker so they aren't run by default. I've had issues with the one I wrote being flaky in the CI builds.

Reply:
Sounds good. how would I do that?

Reply:
Probably need to define a marker in the main `conftest` file (just copy how the other ones are done). Then you just add the decorator:

```python3
@pytest.mark.my_marker
def test_whatever():
    ...
```